src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,157 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final String INDEX_OUT_OF_BOUNDS_MESSAGE = "Index %d "
+            + "does not exist in this LinkedList!";
+    private static final int DEFAULT_DIVISOR = 2;
+    private static final int INDEX_STOPPING_FROM_TAIL = 1;
+    private static final int FIRST_ELEMENT_INDEX = 0;
+    private static final int EMPTY_LIST_VALUE = 0;
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> current = new Node<>(null, value, null);
+        if (tail == null) {
+            tail = current;
+            head = tail;
+        } else {
+            current.prev = tail;
+            current.prev.next = current;
+            tail = current;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkAddIndexExistence(index);
+        Node<T> current;
```

Comment:
You use this variable only in last if-else case. It's better to relocate this invocation. You can optimize your code. You shouldn't always create new current node when we use method in other cases.